tests/unit/aiplatform/test_jobs.py

```diff
@@ -1049,10 +1057,40 @@ def test_update_mdm_job(self, get_mdm_job_mock, update_mdm_job_mock):
         drift_detection_config = aiplatform.model_monitoring.DriftDetectionConfig(
             drift_thresholds=_TEST_MDM_JOB_DRIFT_DETECTION_CONFIG
         )
+        schedule_config = aiplatform.model_monitoring.ScheduleConfig(monitor_interval=1)
+        alert_config = aiplatform.model_monitoring.EmailAlertConfig(
+            user_emails=[_TEST_MDM_USER_EMAIL]
+        )
+        sampling_strategy = aiplatform.model_monitoring.RandomSampleConfig(
+            sample_rate=_TEST_MDM_SAMPLE_RATE
+        )
+        labels = _TEST_MDM_LABEL
+        log_ttl = _TEST_LOG_TTL_IN_DAYS
+        display_name = _TEST_MDM_NEW_NAME
         new_config = aiplatform.model_monitoring.ObjectiveConfig(
             drift_detection_config=drift_detection_config
         )
-        job.update(objective_configs=new_config)
+        job.update(
+            display_name=display_name,
+            schedule_config=schedule_config,
+            alert_config=alert_config,
+            logging_sampling_strategy=sampling_strategy,
+            labels=labels,
+            bigquery_tables_log_ttl=log_ttl,
+            enable_monitoring_pipeline_logs=True,
+            objective_configs=new_config,
+        )
+        gapic_job = job._gca_resource
```

Comment:
It seems like we are testing that the `_gca_resource` is updated but it doesn't seem to be assigned after the update call here: 
https://github.com/googleapis/python-aiplatform/blob/0cdfc9b7cf784317bb810174862d2ea9c2e9497e/google/cloud/aiplatform/jobs.py#L2499-L2504

Reply:
Interesting..I added `print(self._gca_resource)` right before line 2503 and ran both the unit and system tests to double check the contents. Both debug prints successfully showed the updated job contents. I also didn't explicitly run `_sync_resource` on the existing job in the system test and just queried `_gca_resource` directly to verify the update results and it also worked. My guess is that because I'm not decorating the model monitoring methods with `optional_sync()`, the resource is sync'd whenever there is a change.
https://github.com/googleapis/python-aiplatform/blob/0cdfc9b7cf784317bb810174862d2ea9c2e9497e/tests/system/aiplatform/test_model_monitoring.py#L227-L235